### PR TITLE
Allow switching of semester

### PR DESCRIPTION
**Pull request by yangshun**

### CODE REVIEW

File: v3/src/js/views/components/semester-switcher/SemesterSwitcher.jsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import { isValidSemester } from 'utils/timetables';
+
+import styles from './SemesterSwitcher.scss';
+
+type Props = {
+  semester: Semester,
+  onSelectSemester: Function,
+};
+
+class SemesterSwitcher extends Component<Props> {
```
Comment (ZhangYiJiang) [inline v3/src/js/views/components/semester-switcher/SemesterSwitcher.jsx]: This can be a `PureComponent`

**Comment by ZhangYiJiang** [comment]
Ooops, seems like GitHub lost my comment. 

I think it would be useful to have the current semester encoded in the URL. Is this planned for another PR?

**Comment by yangshun** [comment]
I didn't want it to conflict with the semester URL stuff that you were already working on. You want me to add the semester as part of the route? I can do that.

**Comment by ZhangYiJiang** [comment]
@yangshun Yes please - I said that I *wasn't* adding semester to URL in that branch because I didn't know when this feature's PR would land :sweat_smile:

**Comment by yangshun** [comment]
On second thought, I will do it in another PR.

**Pull request merged**